### PR TITLE
feat: added isDocumentRevokable and test cases

### PR DESCRIPTION
**Pull request by isaackps**
# In this PR:
- added a util function: isDocumentRevokable that will return a boolean based on the document if its revokable.

### CODE REVIEW

File: src/shared/utils/__tests__/utils.test.ts [hunk 1]
```diff
@@ -4,21 +4,25 @@ import { SchemaId, WrappedDocument } from "../../../shared/@types/document";
 import * as v2 from "../../../__generated__/schema.2.0";
 import * as v3 from "../../../__generated__/schema.3.0";
 
+const openAttestationDatav2: v2.OpenAttestationDocument & { foo: string } = {
+  issuers: [
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:8]:
```suggestion
const sampleDocumentV2: v2.OpenAttestationDocument & { foo: string } = {
  issuers: [
```

File: src/shared/utils/__tests__/utils.test.ts [hunk 2]
```diff
@@ -75,6 +79,52 @@ describe("Util Functions", () => {
       },
       { version: SchemaId.v3 }
     );
+    wrappedV3DidDocument = await __unsafe__use__it__at__your__own__risks__wrapDocument(
+      {
+        "@context": [
+          "https://www.w3.org/2018/credentials/v1",
+          "https://www.w3.org/2018/credentials/examples/v1",
+          "https://schemata.openattestation.com/com/openattestation/1.0/OpenAttestation.v3.json",
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:87]: able to reduce noise by making use of spread operator to "edit" into variations? so it's clearer what attribute is different/needed in each case. eg:

```
  const sampleDocumentV3: v3.OpenAttestationDocument = {
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://www.w3.org/2018/credentials/examples/v1",
      "https://schemata.openattestation.com/com/openattestation/1.0/OpenAttestation.v3.json",
    ],
    issuer: {
      name: "name",
      type: "OpenAttestationIssuer",
      id: "https://example.com",
    },
    issuanceDate: "2010-01-01T19:23:24Z",
    type: ["VerifiableCredential", "UniversityDegreeCredential", "OpenAttestationCredential"],
    credentialSubject: {
      id: "did:example:ebfeb1f712ebc6f1c276e12ec21",
      degree: {
        type: "BachelorDegree",
        name: "Bachelor of Science and Arts",
      },
    },
    openAttestationMetadata: {
      proof: {
        value: "0xabcf",
        type: v3.ProofType.OpenAttestationProofMethod,
        method: v3.Method.DocumentStore,
      },
      identityProof: {
        identifier: "whatever",
        type: v2.IdentityProofType.DNSTxt,
      },
      template: {
        url: "https://",
        name: "",
        type: v3.TemplateType.EmbeddedRenderer,
      },
    },
    name: "",
    reference: "",
    validFrom: "2010-01-01T19:23:24Z",
  };

  const sampleDocumentV3Did = {
    ...sampleDocumentV3,
    openAttestationMetadata: {
      ...sampleDocumentV3.openAttestationMetadata,
      proof: {
        ...sampleDocumentV3.openAttestationMetadata.proof,
        value: "did:ethr:0xabcf",
        method: v3.Method.Did,
        revocation: {
          type: v3.RevocationType.RevocationStore,
        },
      },
    },
  };

  const sampleDocumentV3TokenRegistry = {
    ...sampleDocumentV3,
    openAttestationMetadata: {
      ...sampleDocumentV3.openAttestationMetadata,
      proof: {
        ...sampleDocumentV3.openAttestationMetadata.proof,
        method: v3.Method.TokenRegistry,
      },
    },
  };
  ```

File: src/shared/utils/__tests__/utils.test.ts [hunk 3]
```diff
@@ -4,60 +4,108 @@ import { SchemaId, WrappedDocument } from "../../../shared/@types/document";
...
+const baseIssuerObjectV2: v2.Issuer = {
+  name: "John",
+  identityProof: {
+    type: v2.IdentityProofType.DNSTxt,
+    location: "tradetrust.io",
+  },
+};
+
+const baseV2DocuemntObject = {
+  version: SchemaId.v2,
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:16]: typo

File: src/shared/utils/__tests__/utils.test.ts [hunk 4]
```diff
@@ -4,60 +4,108 @@ import { SchemaId, WrappedDocument } from "../../../shared/@types/document";
...
-          location: "example.com",
-        },
       },
     ],
   });
+  let v2TransferableDocument: WrappedDocument<any>;
+  let v2VerifiableDocument: WrappedDocument<any>;
+  let v3TransferableDocument: WrappedDocument<v3.OpenAttestationDocument>;
+  let v3VerifiableDocument: WrappedDocument<v3.OpenAttestationDocument>;
+  let v3DidDocument: WrappedDocument<v3.OpenAttestationDocument>;
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:89]: follow the `wrapped` prefix convention? 

```suggestion
  let wrappedV2TransferableDocument: WrappedDocument<any>;
  let wrappedV2VerifiableDocument: WrappedDocument<any>;
  let wrappedV3TransferableDocument: WrappedDocument<v3.OpenAttestationDocument>;
  let wrappedV3VerifiableDocument: WrappedDocument<v3.OpenAttestationDocument>;
  let wrappedV3DidDocument: WrappedDocument<v3.OpenAttestationDocument>;
```

File: src/shared/utils/__tests__/utils.test.ts [hunk 5]
```diff
@@ -4,60 +4,108 @@ import { SchemaId, WrappedDocument } from "../../../shared/@types/document";
...
-            type: "BachelorDegree",
-            name: "Bachelor of Science and Arts",
-          },
-        },
+        ...baseV3DocumentObject,
+      },
+      { version: SchemaId.v3 }
+    );
+    const wrappedV3DidDocument = await __unsafe__use__it__at__your__own__risks__wrapDocument(
+      {
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:99]: why became const?

File: src/shared/utils/__tests__/utils.test.ts [hunk 6]
```diff
@@ -69,33 +117,12 @@ describe("Util Functions", () => {
...
         },
-        name: "",
-        reference: "",
-        validFrom: "2010-01-01T19:23:24Z",
       },
       { version: SchemaId.v3 }
     );
-    wrappedTransferableV3Document = await __unsafe__use__it__at__your__own__risks__wrapDocument(
+    const wrappedTransferableV3Document = await __unsafe__use__it__at__your__own__risks__wrapDocument(
       {
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:124]: same

File: src/shared/utils/__tests__/utils.test.ts [hunk 7]
```diff
@@ -4,60 +4,108 @@ import { SchemaId, WrappedDocument } from "../../../shared/@types/document";
 import * as v2 from "../../../__generated__/schema.2.0";
 import * as v3 from "../../../__generated__/schema.3.0";
 
+const baseIssuerObjectV2: v2.Issuer = {
+  name: "John",
+  identityProof: {
+    type: v2.IdentityProofType.DNSTxt,
+    location: "tradetrust.io",
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:11]:
```suggestion
    location: "https://example.com",
```
explicitly as a sample?

File: src/shared/utils/__tests__/utils.test.ts [hunk 8]
```diff
@@ -289,92 +180,44 @@ describe("Util Functions", () => {
     });
     test("should return all issuers address for 3.0 document", async () => {
       // This test takes some time to run, so we set the timeout to 14s
-      expect(utils.getIssuerAddress(wrappedV3Document)).toStrictEqual("0xabcf");
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:292]: rm comment

File: src/shared/utils/__tests__/utils.test.ts [hunk 9]
```diff
@@ -289,92 +180,44 @@ describe("Util Functions", () => {
...
-        },
-      };
-      expect(utils.getTargetHash(document)).toStrictEqual(
-        "6e3b3b131db956263d142f42a840962d31359fff61c28937d9d1add0ca04c89e"
+      expect(utils.getTargetHash(v3WrappedVerifiableDocument)).toStrictEqual(
+        "797faecb7662eca4e73a82c25a2387d57acd666354c228291fcd664e6aaeb1ba"
       );
     });
     test("should return error when document is not OpenAttestation document", async () => {
       const document: any = {
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:215]:
```suggestion
      const document: WrappedDocument<any> = {
```

File: src/shared/utils/__tests__/utils.test.ts [hunk 10]
```diff
@@ -289,92 +180,44 @@ describe("Util Functions", () => {
...
         signature: {
           type: "SHA3MerkleProof",
-          targetHash: "64b2ed566455d0adbc798a8f824f163d87276dcbd66cacff8a6a4ba28fb800fc",
+          targetHash: "123",
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:218]:
```suggestion
          targetHash: "c5d53262962b192c5c977f2252acd4862f41cc1ccce7e87c5b406905a2726692",
```

sorry lets use back a proper looking hash, the `123` makes me think that incorrect hash matters to return the error for this test, whereas is testing for wrong object shape

File: src/shared/utils/__tests__/utils.test.ts [hunk 11]
```diff
@@ -387,139 +230,91 @@ describe("Util Functions", () => {
...
+      expect(utils.isDocumentRevokable(v3WrappedVerifiableDocument)).toStrictEqual(true);
+    });
+    it("should return false for a v2 transferable document", () => {
+      expect(utils.isDocumentRevokable(v2WrappedTransferableDocument)).toStrictEqual(false);
+    });
+    it("should return false for a v3 transferable document", () => {
+      expect(utils.isDocumentRevokable(v3WrappedTransferableDocument)).toStrictEqual(false);
+    });
+    it("should return true for a v2 DID wrapped document revocation type 'REVOCATION_STORE'", () => {
+      const document: any = {
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:283]:
```suggestion
      const document: WrappedDocument<any> = {
```

File: src/shared/utils/__tests__/utils.test.ts [hunk 12]
```diff
@@ -387,139 +230,91 @@ describe("Util Functions", () => {
...
-          privacy: {
-            obfuscated: [],
+    it("should return false for a v2 DID wrapped document without document store", () => {
+      expect(utils.isDocumentRevokable(v2WrappedDidDocument)).toStrictEqual(false);
+    });
+    it("should return true for a v3 DID wrapped document when revocation store is 'REVOCATION_STORE'", () => {
+      const document: any = {
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:301]:
```suggestion
      const document: WrappedDocument<any> = {
```

File: src/shared/utils/__tests__/utils.test.ts [hunk 13]
```diff
@@ -387,139 +230,91 @@ describe("Util Functions", () => {
...
         },
       };
-      expect(utils.isTransferableAsset(document)).toStrictEqual(false);
+
+      expect(utils.isDocumentRevokable(document)).toStrictEqual(true);
+    });
+    it("should return false for a v3 DID wrapped document when revocation store is 'NONE'", () => {
+      expect(utils.isDocumentRevokable(v3WrappedDidDocument)).toStrictEqual(false);
     });
   });
```
Comment (simboonlong) [inline src/shared/utils/__tests__/utils.test.ts:319]: missing 1 more test case which u mentioned about having a document store but without proof value

File: src/shared/utils/__tests__/utils.test.ts (hunk 8 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:292]: ok

File: src/shared/utils/__tests__/utils.test.ts (hunk 9 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:215]: ok

File: src/shared/utils/__tests__/utils.test.ts (hunk 10 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:218]: ok

File: src/shared/utils/__tests__/utils.test.ts (hunk 11 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:283]: ok

File: src/shared/utils/__tests__/utils.test.ts (hunk 12 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:301]: ok

File: src/shared/utils/__tests__/utils.test.ts (hunk 13 above)
Comment (isaackps) [reply src/shared/utils/__tests__/utils.test.ts:319]: ok will add that in

**Review by simboonlong — APPROVED** [review]
lgtm

**Pull request merged**